### DISCUSSION TITLE
The request signature we calculated does not match the signature you provided. Check your key and signing method.

### DISCUSSION THREAD

**Discussion by Sumis34** [minio/minio#15197] [Q&A]
## NOTE

I have a problem with my proxy. If I set the SERVER_URL env variable for minio i get this error. If i removed it the upload would work but there would be some problems because the server URL is obviously not set anymore

```The request signature we calculated does not match the signature you provided. Check your key and signing method.```

## Expected Behavior
Upload file success

## Current Behavior

cant upload enything because of this error

## Possible Solution

Change Proxy settings

## Your Environment

* Version used (`minio --version`):  latest from [quay.io](https://quay.io/repository/minio/minio)
* Server setup and configuration: see below
* Operating System and version (`uname -a`): Ubuntu 21

Proxy (Apache 2)
```
<VirtualHost *:80>
    ServerName storage.noekrebs.ch
    ServerAlias s3.noekrebs.ch
    ServerAdmin webmaster@localhost

    ErrorLog ${APACHE_LOG_DIR}/storage_error.log
    CustomLog ${APACHE_LOG_DIR}/storage_access.log combined

    ProxyRequests Off
    ProxyVia Block
    ProxyPreserveHost On

    <Proxy *>
         Require all granted
    </Proxy>

    ProxyPass / http://localhost:9000/
    ProxyPassReverse / http://localhost:9000/
</VirtualHost>

<VirtualHost *:443>
    ServerName storage.noekrebs.ch
    ServerAlias s3.noekrebs.ch
    RewriteEngine On
    RewriteCond %{SERVER_NAME} =storage.noekrebs.ch [OR]
    RewriteCond %{SERVER_NAME} =s3.noekrebs.ch
    RewriteRule ^ http://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>
```
How i run minio
![image](https://user-images.githubusercontent.com/63413892/176493461-97f41f2b-c92f-4077-9dde-e586ee46d77a.png)

**Comment by Sumis34** [comment]
Do you need more information to help me with this problem? If you what? Thx for your help :)

**Reply by vadmeste** [reply]
@Sumis34 when you generate the presigned URL while MINIO_SERVER_URL is set, does the presigned url contain MINIO_SERVER_URL in it as well ?

**Reply by Sumis34** [reply]
Hello @vadmeste, As I tried to test my setup with the Minio console im not sure how to verify my presigned URL. Is there any way to see i? 

I have attached a image of the upload and error.

![image](https://user-images.githubusercontent.com/63413892/176940822-b384b150-ff9e-49e8-bb4b-721cff1903b9.png)

If I generate a URL with `client.presignedPutObject` in js and upload a file it works fine.

**Reply by tanhh326** [reply]
I also encountered this problem and was able to resolve it after I turned off the local network proxy. ^_^

**Reply by Alon-Hen-LS** [reply]
@tanhh326 , how did you turn off the local network proxy?

**Reply by James-Oakley** [reply]
Did you solve this?

**Comment by ruempel** [comment]
I get the same message when uploading an object having a no-break space	(U+00A0 aka. character code 160) in the value of the passed user metadata to a bucket.
